aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -17,6 +22,48 @@ namespace {
 #endif
 }
 
+static const std::vector<DispatchKey>& allDispatchKeysInFullSet() {
+  static const auto result = ([]() {
+    std::vector<DispatchKey> vec;
+    for (const auto dispatch_key: DispatchKeySet(DispatchKeySet::FULL)) {
+      vec.push_back(dispatch_key);
+    }
+    return vec;
+  })();
+  return result;
+}
+
+// Returns an array of the same size as the dispatch table, where each
+// entry is the DispatchKey that the corresponding index in the
+// dispatch table represents.
+static const auto& getDispatchTableIndexToKey() {
+  static const auto result = ([]() {
+    using result_type = std::array<DispatchKey, c10::num_runtime_entries>;
+    result_type arr;
+    arr.fill(DispatchKey::Undefined);
+    const auto update_array_entry = [](result_type& arr_, DispatchKey dk) {
+      const auto index = getDispatchTableIndexForDispatchKey(dk);
+      TORCH_INTERNAL_ASSERT(arr_.at(index) == DispatchKey::Undefined);
+      arr_.at(index) = dk;
+    };
+    for (const auto dk_outer: allDispatchKeysInFullSet()) {
+      update_array_entry(arr, dk_outer);
```

Comment:
Why do you need a lambda there?
```suggestion
    for (const auto dk_outer: allDispatchKeysInFullSet()) {
      const auto index = getDispatchTableIndexForDispatchKey(dk);
      TORCH_INTERNAL_ASSERT(arr.at(index) == DispatchKey::Undefined);
      arr.at(index) = dk;
```

Reply:
whoops, used to be called multiple times